torch/distributed/distributed_c10d.py

```diff
@@ -1624,10 +1624,10 @@ def _object_to_tensor(obj, device):
     return byte_tensor, local_size
 
 
-def _tensor_to_object(tensor, tensor_size):
+def _tensor_to_object(tensor, tensor_size, map_location=None):
     tensor = tensor.cpu()
     buf = tensor.numpy().tobytes()[:tensor_size]
-    return _unpickler(io.BytesIO(buf)).load()
+    return torch.load(io.BytesIO(buf),map_location=map_location)
```

Comment:
I don't understand exactly why these changes (torch.load and torch.save) are part of this PR. Shouldn't this PR just be seeking to replace zero's broadcast_object with broadcast_object_list? Not sure if I'm missng something here. cc @awgu

Reply:
I think I used `torch.load()` and `torch.save()` before since they had the `map_location` argument, which is needed to preserve the semantics of ZeRO's `_broadcast_object()`.